src/field_theory/is_alg_closed/classification.lean

```diff
@@ -0,0 +1,219 @@
+... = _ : by simp [max_assoc]
+
+/-- If `K` is an uncountable algebraically closed field, then its
+cardinality is the same as that of a transcendence basis. -/
+lemma cardinal_eq_cardinal_transcendence_basis_of_omega_lt [nontrivial R]
+  (hv : is_transcendence_basis R v) (hR : #R ≤ ω) (hK : ω < #K) : #K = #ι :=
+have ω ≤ #ι,
+  from le_of_not_lt (λ h,
+    not_le_of_gt hK $ calc
+      #K ≤ max (max (#R) (#ι)) ω : cardinal_le_max_transcendence_basis v hv
+     ... ≤ _ : max_le (max_le hR (le_of_lt h)) le_rfl),
+le_antisymm
+  (calc #K ≤ max (max (#R) (#ι)) ω : cardinal_le_max_transcendence_basis v hv
+       ... = #ι : begin
+         rw [max_eq_left, max_eq_right],
+         { exact le_trans hR this },
+         { exact le_max_of_le_right this }
+       end)
+  (mk_le_of_injective (show function.injective v, from hv.1.injective))
+
+end cardinal
+
+variables {K L : Type} [field K] [field L] [is_alg_closed K] [is_alg_closed L]
+
+/-- Two uncountable algebraically closed fields of characteristic zero are isomorphic
+if they have the same cardinality. -/
+@[nolint def_lemma] lemma ring_equiv_of_cardinal_eq_of_char_zero [char_zero K] [char_zero L]
+  (hK : ω < #K) (hKL : #K = #L) : K ≃+* L :=
+begin
+  apply classical.choice,
+  cases exists_is_transcendence_basis ℤ
+    (show function.injective (algebra_map ℤ K),
+      from int.cast_injective) with s hs,
+  cases exists_is_transcendence_basis ℤ
+    (show function.injective (algebra_map ℤ L),
+      from int.cast_injective) with t ht,
+  have : #s = #t,
+  { rw [← cardinal_eq_cardinal_transcendence_basis_of_omega_lt _ hs (le_of_eq mk_int) hK,
+        ← cardinal_eq_cardinal_transcendence_basis_of_omega_lt _ ht (le_of_eq mk_int), hKL],
+    rwa ← hKL },
+  cases cardinal.eq.1 this with e,
+  exact ⟨equiv_of_transcendence_basis _ _ e hs ht⟩
+end
+
+private lemma ring_equiv_of_cardinal_eq_of_char_p (p : ℕ) [fact p.prime]
+  [char_p K p] [char_p L p] (hK : ω < #K) (hKL : #K = #L) : K ≃+* L :=
+begin
+  apply classical.choice,
+  cases exists_is_transcendence_basis (zmod p)
+    (show function.injective (algebra_map (zmod p) K),
+      from ring_hom.injective _) with s hs,
+  cases exists_is_transcendence_basis (zmod p)
+    (show function.injective (algebra_map (zmod p) L),
+      from ring_hom.injective _) with t ht,
+  have : #s = #t,
+  { rw [← cardinal_eq_cardinal_transcendence_basis_of_omega_lt _ hs
+      (le_of_lt $ lt_omega_iff_fintype.2 ⟨infer_instance⟩) hK,
+        ← cardinal_eq_cardinal_transcendence_basis_of_omega_lt _ ht
+      (le_of_lt $ lt_omega_iff_fintype.2 ⟨infer_instance⟩), hKL],
+    rwa ← hKL },
+  cases cardinal.eq.1 this with e,
+  exact ⟨equiv_of_transcendence_basis _ _ e hs ht⟩
+end
+
+/-- Two uncountable algebraically closed fields are isomorphic
+if they have the same cardinality and the same characteristic. -/
+@[nolint def_lemma] lemma ring_equiv_of_cardinal_eq_of_char_eq (p : ℕ) [char_p K p] [char_p L p]
```

Comment:
I know this has already merged, but are these `lemma`s instead of `def`s?

Reply:
It's just a random isomorphism chosen using the axiom of choice so you never want to unfold the definition.

Reply:
I made [a Zulip thread](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/.40.5Bnolint.20def_lemma.5D.20lemma/near/273793990) about this.